### DISCUSSION TITLE
Override Core Method correctly?

### DISCUSSION THREAD

**Discussion by rbartholomay** [adonisjs/core#1460] [Help] [answered]
Hi there,

in the [docu ](https://preview.adonisjs.com/blog/introducing-adonisjs-v5/#extending-adonisjs-core)i find the example how to extend the core (Response)
In my case i need the correct way to to this for the Logger.log method

How can i override this method that is does exactly the same as before with an extra console.log for example (senseless i know)

Bye, René

**Comment by smickelbeard** [comment, marked as answer]
Maybe use `apply` to extend the Logger's function?

IE, in a AppProvider:

```typescript
export default class AppProvider {
  constructor (protected container: IocContract) {}

  public async boot () {
    const Logger = (await import('@ioc:Adonis/Core/Logger')).default
    const oldInfo = Logger.info
    const extendedLogger = function() {
      oldInfo.apply(this, arguments)
      console.log('Bar')
    }
    Logger.info = extendedLogger
    Logger.info('Foo') // outputs Foo using original logger and Bar using console.log
  }
}
```

**Reply by rbartholomay** [reply]
Nice!  Thank your very much!
